src/s3_path.rs

```diff
@@ -54,10 +54,7 @@ pub enum S3PathErrorKind {
 
 /// See [bucket nameing rules](https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html#bucketnamingrules)
 fn check_bucket_name(name: &str) -> bool {
-    /// helper function
-    fn is_digit_or_lowercase(b: u8) -> bool {
-        b.is_ascii_lowercase() || b.is_ascii_digit()
-    }
+    let is_digit_or_lowercase = |b: u8| -> bool { b.is_ascii_lowercase() || b.is_ascii_digit() };
```

Comment:
it seems assign closure to a variable is not recommended in Rust

Reply:
`is_ascii_lowercase` and `is_ascii_digit` are not yet stable as `const fn`

<https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4bf93cdabb9a75b8ad49f99470d2e34a>

but clippy will complain: this could be a `const fn`

Reply:
if `is_digit_or_lowercase` is a `fn`, one of `cargo check` and `cargo clippy` will fail

Reply:
what's the error msg?

Reply:
or directly use the closure, instead of assign the closure to a variable?